### DISCUSSION TITLE
Header 'Date' ignored inside request headers

### DISCUSSION THREAD

**Discussion by gs1-rob-392** [fastapi/fastapi#6216] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Depends

async def check_access(request: Request, date: str | None = Header(default=None)):
    print(date)
    print(request.headers)

app = FastAPI(
    dependencies=[Depends(check_access)],
)

@app.get("/")
async def read_root():
    return {"Hello": "World"}
```

### Description

Hello,
I'm trying to add Date header to my request from swagger field, but every string is ignored.

- The first print is: None
- The second print not include any Date header.
- Inside curl request the UI shows -H parameter with date: '{my_string}' but it seems to be ignored.

If I change the name from date to anything else (dates  for example), the header works properly.
Do you have any solution to include and read date header???

### Operating System

macOS, Other

### Operating System Details

Docker container

### FastAPI Version

0.87.0

### Python Version

3.10.5

**Comment by jgould22** [comment]
Date is a forbidden header name in fetch, which is likely what is being used in the OpenAPI UI (which I assume is what you are referring to) https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Date

Does it do the same thing if you use Curl directly ?

**Comment by ysnghr** [comment]
You can't. Since you are sending a request it MUST be terminated according to the standard when you set a whole list of prohibited headers:

[https://www.w3.org/TR/2010/CR-XMLHttpRequest-20100803/](https://www.w3.org/TR/2010/CR-XMLHttpRequest-20100803/)

Terminate these steps if header is a case-insensitive match for one of the following headers:

- Accept-Charset
- Accept-Encoding
- Connection
- Content-Length
- Cookie
- Cookie2
- Content-Transfer-Encoding
- Date
- Expect
- Host
- Keep-Alive
- Referer
- TE
- Trailer
- Transfer-Encoding
- Upgrade
- User-Agent
- Via

**Comment by gs1-rob-392** [comment]
> Date is a forbidden header name in fetch, which is likely what is being used in the OpenAPI UI (which I assume is what you are referring to) https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Date
> 
> Does it do the same thing if you use Curl directly ?

If I use a Curl, date header is readed correctly, maybe it is a bug of swagger documentation that doesn't pass prohibited headers correctly.

**Comment by iudeen** [comment]
@gs1-rob-392 how will it be a bug when it does something that is expected?🤔 
I think its expected behavior.

Suggestion: you can use another name for your date header, something like `X-Date`

**Comment by caeser1996** [comment]
@gs1-rob-392
`from fastapi import FastAPI, Request

app = FastAPI()


@app.get("/")
async def root(request: Request):
    date= request.headers.get('date')
    return {"date": date}`
Can you try doing the above and see if you can get the date and then do the same thing on the middleware?

**Comment by jgould22** [comment, marked as answer]
> @gs1-rob-392 how will it be a bug when it does something that is expected?thinking I think its expected behavior.
> 
> Suggestion: you can use another name for your date header, something like `X-Date`

I mean it would probably be nice if the UI warned the user or something if they are trying to use a forbidden header.  The fact that is is forbidden by fetch but is accommodated by curl can be sorta confusing.

Never the less that would be something to be solved by https://github.com/swagger-api/swagger-ui it think, not FastAPI

**Comment by gs1-rob-392** [comment]
Swagger bug
